### PR TITLE
Added emissionAngle function to SensorUtils library

### PR DESCRIPTION
**Pull request by twilson271828**
Also added two new tests.

### CODE REVIEW

**Comment by jlaura** [comment]
@AstroKEW78 Is the CI passing? Does the code have tests?

File: include/sensorutils/SensorUtils.h [hunk 1]
```diff
@@ -5,5 +5,8 @@
 using namespace std;
 
 double PhaseAngle(vector<double> & instPosition, vector<double> & sunPosition);
+double EmissionAngle(const std::vector<double>  &observerBodyFixedPosition,
```
Comment (makaylas) [inline include/sensorutils/SensorUtils.h]: We're already using namespace std you don't need the std::

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.10)
 set(COVERAGE OFF CACHE BOOL "Coverage")
 
 # Link runSensorUtilsTests with what we want to test and the GTest and pthread library
-add_executable(runSensorUtilsTests TestyMcTestFace.cpp)
+add_executable(runSensorUtilsTests SensorUtilsTesting.cpp)
```
Comment (makaylas) [inline tests/CMakeLists.txt]: You changed the .cpp. Where is the new .cpp? That is why the CI is failing

Comment (twilson271828) [reply tests/CMakeLists.txt]: I forgot to add it to this PR.  I just committed it.

File: include/sensorutils/SensorUtils.h (hunk 1 above)
Comment (twilson271828) [reply include/sensorutils/SensorUtils.h]: So we are.  I've deleted the std:: stuff.

File: tests/SensorUtilsTesting.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+   vector<double> surfaceNormal3{1.0,0.0,0.0};
+   EXPECT_NEAR(0.0,EmissionAngle(observerBodyFixedPosition3, groundPtIntersection3,surfaceNormal3),1e-5);
+
+
+   vector<double> observerBodyFixedPosition4{1.0,1.0,1.0};
+   vector<double> groundPtIntersection4{0.0,0.0,0.0};
+   vector<double> surfaceNormal4{-2.0,-2.0,2.0};
+   EXPECT_NEAR(3.141592653589793,EmissionAngle(observerBodyFixedPosition4, groundPtIntersection4,surfaceNormal4),1e-5);
+
+
```
Comment (jlaura) [inline tests/SensorUtilsTesting.cpp]: Super minor: Can you comment on what these are testing.  In other words, what is the purpose of each test. I suspect that the next person that comes along will appreciate knowing what edge case (or not) each test is designed to cover.

**Comment by jlaura** [comment]
@makaylas Can you update your review on this to indicate that everything looks good or that a change is desired?  Thanks!

File: src/SensorUtils.cpp [hunk 4]
```diff
@@ -2,8 +2,55 @@
 
 #include <armadillo>
 #include <iostream>
+#include <iomanip>
```
Comment (jlaura) [inline src/SensorUtils.cpp]: What is this being used by?

**Comment by twilson271828** [comment]
Kaj is correct..there is a typedef in Armadillo which defines both vec and dvec as Col (column vectors of type double).  The Col class being a subclass of the Mat class.

File: include/sensorutils/SensorUtils.h [hunk 5]
```diff
@@ -6,4 +6,8 @@ using namespace std;
 
 double PhaseAngle(const vector<double> & instPosition, const vector<double> & sunPosition, const vector<double> & surfaceIntersection);
 
+double EmissionAngle(const vector<double>  &observerBodyFixedPosition,
+                     const vector<double> &groundPtIntersection,
+                     const vector<double> &surfaceNormal);
```
Comment (jessemapel) [inline include/sensorutils/SensorUtils.h]: Once we get a few more of these done, we should come up with some standardized parameter names (ie. what do we call an intersection point, the sun position, etc)

File: tests/SensorUtilsTesting.cpp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+   vector<double> observerBodyFixedPosition4{1.0,1.0,1.0};
+   vector<double> groundPtIntersection4{0.0,0.0,0.0};
+   vector<double> surfaceNormal4{-2.0,-2.0,2.0};
+   EXPECT_NEAR(M_PI,EmissionAngle(observerBodyFixedPosition4, groundPtIntersection4,surfaceNormal4),1e-5);
+
+
+
+}
```
Comment (jessemapel) [inline tests/SensorUtilsTesting.cpp]: I would really like to see these test broken up into multiple tests. That way, if a specific case fails, we know exactly what it was from the ctest output instead of having to try and read the gtest output.
